client/src/main/java/org/apache/rocketmq/schema/registry/client/CachedSchemaRegistryClient.java

```diff
@@ -0,0 +1,485 @@
+    }
+
+    @Override
+    public RegisterSchemaResponse registerSchema(String subject, String schemaName,
+        RegisterSchemaRequest request) throws RestClientException, IOException {
+        return restService.registerSchema(subject, schemaName, request);
+    }
+
+    @Override
+    public RegisterSchemaResponse registerSchema(String clusterName, String tenant, String subjectName,
+        String schemaName, RegisterSchemaRequest request) throws IOException, RestClientException {
+        return restService.registerSchema(clusterName, tenant, subjectName, schemaName, request);
+    }
+
+    @Override
+    public DeleteSchemeResponse deleteSchema(String cluster, String tenant,
+        String subject) throws IOException, RestClientException {
+        String subjectFullName = String.format("%s/%s/%s", cluster, tenant, subject);
+
+        schemaCacheBySubject.invalidate(subjectFullName);
+        //invalidate schemaCacheBySubjectAndVersion
+        if (subjectToVersion.get(subjectFullName) != null) {
+            subjectToVersion.get(subjectFullName).forEach(
+                version -> schemaCacheBySubjectAndVersion.invalidate(new SubjectAndVersion(cluster, tenant, subject, version)));
+        }
+        //invalidate schemaCacheBySubjectAndId
+        if (subjectToId.get(subjectFullName) != null) {
+            subjectToId.get(subjectFullName).forEach(
+                recordId -> schemaCacheBySubjectAndId.invalidate(new SubjectAndId(cluster, tenant, subject, recordId)));
+        }
+        // invalidate schemaCache
+        if (subjectToSchema.get(subjectFullName) != null) {
+            subjectToSchema.get(subjectFullName).forEach(
+                schema -> schemaCache.invalidate(new SubjectAndSchema(cluster, tenant, subject, schema)));
+        }
+        subjectToVersion.remove(subjectFullName);
+        subjectToId.remove(subjectFullName);
+        subjectToSchema.remove(subjectFullName);
+        return restService.deleteSchema(cluster, tenant, subject);
+    }
+
+    @Override
+    public DeleteSchemeResponse deleteSchema(String cluster, String tenant, String subject,
+        long version) throws IOException, RestClientException {
+        String subjectFullName = String.format("%s/%s/%s", cluster, tenant, subject);
+        schemaCacheBySubject.invalidate(subjectFullName);
+        schemaCacheBySubjectAndVersion.invalidate(new SubjectAndVersion(cluster, tenant, subject, version));
+        subjectToVersion.get(subjectFullName).remove(version);
+
+        return restService.deleteSchema(cluster, tenant, subject, version);
+    }
+
+    @Override
+    public UpdateSchemaResponse updateSchema(String subject, String schemaName,
+        UpdateSchemaRequest request) throws RestClientException, IOException {
+        // invalidate schemaCache
+        schemaCache.invalidate(new SubjectAndSchema(DEFAULT_CLUSTER, DEFAULT_TENANT, subject, schemaName));
+        return restService.updateSchema(subject, schemaName, request);
```

Comment:
There can also update schemaCache after update schema success directly.

Reply:
Same as regitster, I think some fileds such as dependency and type cannot be find in UpdateSchemaResponse or UpdateSchemaRequest, so it's hard to make a GetSchemaResponse without querying